envconfig_test.go

```diff
@@ -70,6 +75,7 @@ type EmbeddedButIgnored struct {
 
 func TestProcess(t *testing.T) {
 	var s Specification
+	s.NestedInterface = &Anonymous{}
```

Comment:
this seems backwards -- you could embed the generic/shared configuration struct in the specific one, then you don't need the `interface{}` placeholder.